### PR TITLE
[15.0][MIG] component

### PR DESCRIPTION
**Pull request by ajaniszewska-dev**

### CODE REVIEW

File: component/i18n/am.po [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Translation of Odoo Server.
```
Comment (simahawk) [inline component/i18n/am.po]: these translation files can be trashed IMO, not mandatory of course :)

File: component/tests/common.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+        # super)
+        common.TransactionCase.setUp(self)
+        ComponentMixin.setUp(self)
+        # There's no env on setUpClass of TransactionCase, must do it here.
+        self.env.context = dict(
+            self.env.context, components_registry=self._components_registry
+        )
+
+
+class SavepointComponentCase(common.SavepointCase, ComponentMixin):
```
Comment (simahawk) [inline component/tests/common.py]: This class should be deprecated as common.SavepointCase.
I would make it inherit from `TransactionComponentCase` directly and just log deprecation.
Setup code should be moved from setUp to setUpClass.

File: component/tests/common.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+        common.TransactionCase.setUp(self)
+        ComponentRegistryCase._setup_registry(self)
+        self.collection = self.env["collection.base"]
+
+    def tearDown(self):
+        ComponentRegistryCase._teardown_registry(self)
+        common.TransactionCase.tearDown(self)
+
+
+class SavepointComponentRegistryCase(common.SavepointCase, ComponentRegistryCase):
```
Comment (simahawk) [inline component/tests/common.py]: same

File: component/tests/common.py [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+    Or::
+
+        collection_record = self.env['my.collection'].browse(1)
+        with collection_record.work_on(
+                'res.partner',
+                components_registry=self.comp_registry) as work:
+
+    """
+
+    if sys.version_info < (3, 8):
```
Comment (simahawk) [inline component/tests/common.py]: this should be trashed if we don't run on 3.7 anymore

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by simahawk** [comment]
pls check pre-commit too

File: component/tests/test_build_component.py [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+
+        Component1._build_component(self.comp_registry)
+        msg = "Component.*already exists.*"
+        with self.assertRaisesRegex(TypeError, msg):
+            Component2._build_component(self.comp_registry)
+
+    def test_no_parent(self):
+        """Ensure we can't _inherit a non-existent component"""
+
+        class Component1(Component):
```
Comment (ajaniszewska-dev) [inline component/tests/test_build_component.py]: `component/tests/test_build_component.py:59: [R7980(consider-merging-classes-inherited), TestBuildComponent.test_inherit_bases.Component2] Consider merging classes inherited to "component1" from test_build_component.py:62, test_build_component.py:184, test_build_component.py:269.
`

@simahawk  pre commit complains for that as above. Possible to disable it somehow?

Comment (simahawk) [reply component/tests/test_build_component.py]: in this case just customize the class w/ a silly method or attribute that is never used, eg:

```
class Component1(Component):
    def say_hello(self):
        """Dumb method."""
        return "hello"
```

Comment (ajaniszewska-dev) [reply component/tests/test_build_component.py]: @simahawk  thanks, but I tried both this methods and params, to me it doesn't work.
`# pylint: disable=R7980` above affected classes maybe?

Comment (simahawk) [reply component/tests/test_build_component.py]: Checked. Just do this: 
1. rollback your changes to address this
2. add this block after the file header

```py
# Copyright 2017 Camptocamp SA
# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html)

# Tell pylint to not bother us for all our fake component classes
# pylint: disable=consider-merging-classes-inherited
```

**Comment by simahawk** [comment]
@ajaniszewska-dev if you delete commits w/ push force, the comment done on specific commit lose reference. Just use fixup for now :pray:

**Comment by simahawk** [comment]
> @ajaniszewska-dev if you delete commits w/ push force, the comment done on specific commit lose reference. Just use fixup for now pray

or maybe is github is just crazy here because is the 1st time I see this messed up summary of commits :thinking:

**Comment by simahawk** [comment]
@ajaniszewska-dev there are still things to be fixed. Could you please propose a PR to add the module as not installable first? We can fast track it and then we'll have an easier review here.

**Comment by lmignon** [comment]
To ease the migration and avoid all this mess with the commit histrory, I propose to recreate the 15.0 branch from the 14.0 with all the addons declared with `"installable": False. What do you think about @simahawk @guewen @yankinmax

**Comment by simahawk** [comment]
@lmignon I was thinking the same but I had some issues w/ my MT setup and I had no time yesterday.
I'll see if I can do it today, unless any of you can make it before :wink:

**Comment by lmignon** [comment]
@simahawk I'll do it into the coming hours

**Comment by lmignon** [comment]
@yankinmax @simahawk @guewen 15.0 is now initialized from 14.0...

**Comment by dreispt** [comment]
@liweijie0812 @simahawk @guewen I updated https://github.com/OCA/connector/issues/418 accordingly.

**Pull request closed without merging**